e2e_tests/tests/cluster/test_groups.py

```diff
@@ -0,0 +1,132 @@
+import json
+import subprocess
+from typing import Any, List
+
+import pytest
+
+from tests import config as conf
+
+from .test_users import get_random_string
+
+
+def det_cmd(cmd: List[str]) -> subprocess.CompletedProcess:
+    return subprocess.run(
+        ["det", "-m", conf.make_master_url()] + cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+
+
+def det_cmd_json(cmd: List[str]) -> Any:
+    res = det_cmd(cmd)
+    assert res.returncode == 0
+    return json.loads(res.stdout)
+
+
+def det_cmd_expect_error(cmd: List[str], expected: str) -> None:
+    res = det_cmd(cmd)
+    assert res.returncode != 0
+    assert expected in res.stderr.decode()
+
+
+@pytest.mark.e2e_cpu
+@pytest.mark.parametrize("add_users", [[], ["admin", "determined"]])
+def test_group_creation(add_users: List[str]) -> None:
+    group_name = get_random_string()
+    create_group_cmd = ["user-group", "create", group_name]
+    for add_user in add_users:
+        create_group_cmd += ["--add-user", add_user]
+
+    assert det_cmd(create_group_cmd).returncode == 0
+
+    # Can view through list.
+    group_list = det_cmd_json(["user-group", "list", "--json"])
+    assert sum([group["group"]["name"] == group_name for group in group_list["groups"]]) == 1
```

Comment:
and if you want to check if it's really "only one" I'd think `len` is few characters longer but I think reflects the purpose better?
```suggestion
    assert len([group for group in group_list["groups"] if group["group"]["name"] == group_name]) == 1
```